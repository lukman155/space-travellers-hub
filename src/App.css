nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 2rem;
  background-color: #fff;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
}

nav > div {
  display: flex;
  align-items: center;
  gap: 15px;
}

.logo {
  width: 50px;
  height: 50px;
}

.nav-list {
  display: flex;
  list-style: none;
  gap: 15px;
}

a {
  text-decoration: none;
  color: #007bff;
}

a:hover {
  text-decoration: underline;
}

.underline {
  text-decoration: underline;
}

.vertical-line {
  border-left: 2px solid black;

  /* margin-left: 5px; */
}
