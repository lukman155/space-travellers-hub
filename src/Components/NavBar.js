import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../Images/planet.png';

const NavBar = () => {
  const activeStyle = {
    textDecoration: 'underline',
  };

  const activeClassName = 'underline';

  return (
    <nav>
      <div>
        <NavLink
          to="/"
        >
          <img className="logo" src={logo} alt="logo" />
        </NavLink>
        <h1>Space Travelers Hub</h1>
      </div>
      <ul className="nav-list">
        <li>
          <NavLink
            to="rockets"
            style={({ isActive }) => (isActive ? activeStyle : undefined)}
          >
            Rockets
          </NavLink>
        </li>
        <li>
          <NavLink
            to="missions"
            className={({ isActive }) => (isActive ? activeClassName : undefined)}
          >
            Missions
          </NavLink>

        </li>
        <div className="vertical-line" />
        <li>
          <NavLink to="myprofile">
            {({ isActive }) => (
              <span
                className={
                  isActive ? activeClassName : undefined
                }
              >
                My Profile
              </span>
            )}
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};
export default NavBar;
