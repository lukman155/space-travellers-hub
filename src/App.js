import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Missions from './Pages/Missions';
import MyProfile from './Pages/MyProfile';
import Rockets from './Pages/Rockets';
import NavBar from './Components/NavBar';
import './App.css';

const App = () => (
  <div className="container">
    <NavBar />
    <Routes>
      <Route exact path="/" element={<Rockets />} />
      <Route path="rockets" element={<Rockets />} />
      <Route path="missions" element={<Missions />} />
      <Route path="myprofile" element={<MyProfile />} />
    </Routes>
  </div>
);

export default App;
