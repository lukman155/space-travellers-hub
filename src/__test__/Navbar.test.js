import React from 'react';
import { fireEvent, screen, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../Redux/configureStore';
import App from '../App';
import NavBar from '../Components/Navbar';

describe('Navbar', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
    );
    fireEvent.click(screen.getByText(/My Profile/));

    expect(screen.getByText(/My Missions/)).toBeInTheDocument();
  });
});
