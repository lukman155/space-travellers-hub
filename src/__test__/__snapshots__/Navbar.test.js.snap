// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav>
  <div>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo"
        src="planet.png"
      />
    </a>
    <h1>
      Space Travelers Hub
    </h1>
  </div>
  <ul
    className="nav-list"
  >
    <li>
      <a
        className=""
        href="/rockets"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <div
      className="vertical-line"
    />
    <li>
      <a
        className=""
        href="/myprofile"
        onClick={[Function]}
      >
        <span>
          My Profile
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
